R/azure.R

```diff
@@ -0,0 +1,147 @@
+        local_file_path
+      ),
+      error = function(con) {
+        cli::cli_abort(
+          c(
+            "Error downloading blob {.path {blob}}",
+            "Using container {.path {container_name}}",
+            "Writing to local file path {.path local_file_path}"
+          ),
+          parent = con
+        )
+      }
+    )
+  }
+  cli::cli_alert_success("Blobs {.path {blob_names}} downloaded successfully")
+  invisible(NULL)
+}
+
+download_file_from_container <- function(
+    blob_storage_path,
+    container,
+    local_file_path) {
+  cli::cli_alert_info(
+    "Downloading blob {.path {blob_storage_path}} to {.path {local_file_path}}"
+  )
+
+  AzureStor::download_blob(
+    container = container,
+    src = blob_storage_path,
+    dest = local_file_path,
+    overwrite = TRUE
+  )
+
+  cli::cli_alert_success(
+    "Blob {.path {blob_storage_path}} downloaded successfully"
+  )
+
+  invisible(local_file_path)
+}
+
+#' Load Azure Blob endpoint using credentials in environment variables
+#'
+#' This **impure** function depends on the environment variables:
+#' * TENANT_ID
+#' * SUBSCRIPTION
+#' * RESOURCE_GROUP
+#' * STORAGE_ACCOUNT
+#'
+#' It will error out if any of the above is not set.
+#' @param container_name The Azure Blob Storage container associated with the
+#'   credentials
+#' @return A Blob endpoint
+#' @export
+fetch_blob_container <- function(container_name) {
+  cli::cli_alert_info(
+    "Attempting to connect to container {.var {container_name}}"
+  )
+  cli::cli_alert_info("Loading Azure credentials from env vars")
+  # nolint start: object_name_linter
+  TENANT_ID <- fetch_credential_from_env_var("TENANT_ID")
+  SUBSCRIPTION <- fetch_credential_from_env_var("SUBSCRIPTION")
+  RESOURCE_GROUP <- fetch_credential_from_env_var("RESOURCE_GROUP")
+  STORAGE_ACCOUNT <- fetch_credential_from_env_var("STORAGE_ACCOUNT")
+  # nolint end: object_name_linter
+  cli::cli_alert_success("Credentials loaded successfully")
+
+
+  cli::cli_alert_info("Authenticating with loaded credentials")
+  az <- AzureRMR::get_azure_login(TENANT_ID)
+  subscription <- az$get_subscription(SUBSCRIPTION)
+  resource_group <- subscription$get_resource_group(RESOURCE_GROUP)
+  storage_account <- resource_group$get_storage_account(STORAGE_ACCOUNT)
+
+  # Getting the access key
+  keys <- storage_account$list_keys()
+  access_key <- keys[["key1"]]
+
+  endpoint <- AzureStor::blob_endpoint(
+    storage_account$properties$primaryEndpoints$blob,
+    key = access_key
+  )
+
+  container <- AzureStor::storage_container(endpoint, container_name)
+  cli::cli_alert_success("Authenticated connection to {.var {container_name}}")
+
+  return(container)
+}
+
+#' Fetch Azure credential from environment variable
+#'
+#' And throw an informative error if credential is not found
+#'
+#' @param env_var A character, the credential to fetch
+#'
+#' @return The associated value
+#' @export
+fetch_credential_from_env_var <- function(env_var) {
```

Comment:
This was throwing a "con not defined" error.

Reply:
https://github.com/CDCgov/cfa-epinow2-pipeline/pull/5/commits/3d1b053d20d86a75536649add97687c048a76a8e